fix(profiling): Add details around frames and validation (#872)
getsentry/develop: src/docs/sdk/profiling-format-sampled.mdx

Stacks should be ordered from outermost to innermost frame.

> It's encouraged to deduplicate stacks by only storing it once and referring to

Reviewer: ```suggestion
It's encouraged to de-duplicate stacks by only storing them once and referring to
```

- no transaction associated with the profile
- any of the metadata required is missing
- size of the profile exceeds 50MB

> - a profile exceeds 30 seconds (difference between last and first sample

Reviewer: ```suggestion
- a profile exceeds 30 seconds (delta between last and first sample
```

  timestamp)

It is suggested to validate these conditions before sending the profile.

It's suggested to remove unnecessary data like thread data without samples from
`thread_metadata`.

## Ingestion

> After this payload is generated, serialize it as JSON, pack it into the same
> envelope as the associated transaction with the `ItemType` `"profile"` and send
> it to Relay.

Reviewer: ```suggestion
After the payload has been generated, serialize it into JSON and attach it as an `"type":"profile"` item to the same envelope as the associated transaction.
```
